server/services/converters.go

```diff
@@ -27,6 +30,28 @@ func blockSummaryToIdentifier(blockSummary *resources.BlockSummary) *types.Block
 	}
 }
 
+func blockIdentifierToAccountQueryOptions(identifier *types.PartialBlockIdentifier) (resources.AccountQueryOptions, error) {
+	if identifier == nil {
+		return resources.AccountQueryOptions{OnFinalBlock: true}, nil
+	}
+
+	if identifier.Index != nil {
+		blockNonce := common.OptionalUint64{Value: uint64(*identifier.Index), HasValue: true}
```

Comment:
multiline here and on L49 & L52?
Optional: we might have constructors for AccountQueryOptions instances

Reply:
Fixed. Also added constructors (but kept the DTO exported, for simplicity, though).